### ISSUE TITLE
Generic low-level field wrapper component

### ISSUE THREAD

**Issue by web-padawan** [vaadin/vaadin-core#238]
Originally `vaadin-custom-field` was designed to be a wrapper that provides a markup (label, error message) and styles to create your own field component.

However, it was implemented so that is also handles validation, supports multiple inputs etc.

We still need a "dumb" component that would only provide slots and styles for:

- label
- error message
- helper text

Same as `vaadin-overlay`, that would be purely internal component (no Java API).
In future, it could be reused by `vaadin-text-field-container` vaadin/vaadin-text-field#67

**Comment by jouni** [comment]
In my opinion, this component should not be completely “dumb”, not just about slots and styles. I think it should handle accessibility as well, manage the label, description, and validation message associations, as well as handle showing the required field indicator if the associated/wrapped field is marked as required.

**Comment by lightoze** [comment]
Please also take into account a use case when a user might want to use the wrapper for a custom component to make it appear like a field, but not being a field in strict terms (i.e. not having a distinct value).

**Comment by jouni** [comment]
Yeah, that is a good point as well, that I’ve also been thinking about. The wrapped component/element could be a plain string, for example, which would make it easy to create a “read-only” version of a form, without resorting to the read-only mode of the input components.

**Comment by lightoze** [comment]
Exactly. Another example from personal experience is wrapping several caption-less fields, which are then bound individually in Java.

**Comment by jouni** [comment]
Heh, the hits keep coming. That’s also another thing that I’ve been considering, having a single field contain multiple inputs. That’s basically what the current `<vaadin-custom-field>` component tries to do, but it’s not very thought through.

**Comment by lightoze** [comment]
At the moment I use a custom theme (in addition to Lumo) for standard fields to produce a result similar to vaadin-form-item but with the following advantages:
- not having to change field initialization code
- not having some of critical vaadin-form-item bugs (https://github.com/vaadin/vaadin-form-layout/issues/122)
- being able to switch behaviour based on screen size.
But especially when it comes to aligning multiple fields and their labels, I don't think this solution is very flexible.

So it came to my mind that this generic field wrapper could have an option to have a reference to external label element (label element ID would suffice) and update its properties (inline-html for label text, required, error, readonly attributes...) instead of built-in label. This is complicated, but I think could realistically be done and it would make things a lot more flexible when it comes to field label layout.

Examply:
```html
<vaadin-field-label id="abc"></vaadin-field-label>
...
<vaadin-text-field label-id="abc" label="Text" value="test" required></vaadin-field-label>
```

After initializing text field component would update the label to:
```html
<vaadin-field-label id="abc" required has-value>Text</vaadin-field-label>
```
